app/parse.py

```diff
@@ -15,9 +37,81 @@ class Product:
     num_of_reviews: int
 
 
+_driver: WebDriver | None = None
+
+
+def get_driver() -> WebDriver:
+    return _driver
+
+
+def set_driver(new_driver: WebDriver) -> None:
+    global _driver
+    _driver = new_driver
+
+
+def parse_single_product(product_soup: BeautifulSoup) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(".description").text,
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=int(len(product_soup.select(".glyphicon"))),
+        num_of_reviews=int(
+            product_soup.select_one(".ratings > p").text.split()[0]
+        ),
+    )
+
+
+def get_page_soup(url: str) -> list[Product]:
+    page = requests.get(url).content
+    page_soup = BeautifulSoup(page, "html.parser")
+
+    if page_soup.select_one(".ecomerce-items-scroll-more"):
+        with webdriver.Chrome() as new_driver:
+            set_driver(new_driver)
+            driver = get_driver()
+            driver.get(url)
+            button = driver.find_element(
+                By.CLASS_NAME, "ecomerce-items-scroll-more"
+            )
+            driver.find_element(By.CLASS_NAME, "acceptCookies").click()
+
+            while not button.get_property("style"):
+                button.click()
+                time.sleep(0.1)
+
+                button = driver.find_element(
+                    By.CLASS_NAME, "ecomerce-items-scroll-more"
+                )
+
+            with open("index.html", "w", encoding="cp1251") as file:
+                file.write(driver.page_source)
+
+        with open("index.html") as file:
+            page = file.read().replace("&nbsp;", " ")
+        os.remove("index.html")
+
+        page_soup = BeautifulSoup(page, "html.parser")
+        return [parse_single_product(page)
+                for page in page_soup.select(".thumbnail")]
+
+    page_soup = page_soup.select(".thumbnail")
+    return [parse_single_product(page) for page in page_soup]
+
+
+def write_products_to_csv(name: str, products: [Product]) -> None:
+    product_fields = [field.name for field in fields(Product)]
+    with open(name, "w", newline="", encoding="cp1251") as file:
```

Comment:
Why do you use this encoding?

Reply:
Because I write from under Windows, is it better not to specify the encoding?